lib/core-integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs

```diff
@@ -882,6 +889,102 @@ spec = describe "NEW_SHELLEY_TRANSACTIONS" $ do
             , expectResponseCode HTTP.status202
             , expectField (#coinSelection . #inputs) (`shouldSatisfy` (not . null))
             ]
+
+    it "TRANS_NEW_SIGN_01 - Sign single-output transaction" $ \ctx -> runResourceT $ do
+        w <- fixtureWallet ctx
+        let amt = minUTxOValue (_mainEra ctx)
+
+        -- Construct tx
+        payload <- mkTxPayload ctx w amt
+        let constructEndpoint = Link.createUnsignedTransaction @'Shelley w
+        sealedTx <- getFromResponse #transaction <$>
+            request @(ApiConstructTransaction n) ctx constructEndpoint Default payload
+
+        -- Sign tx
+        let toSign = Json [json|
+                { "transaction": #{sealedTx}
+                , "passphrase": #{fixturePassphrase}
+                }|]
+        let signEndpoint = Link.signTransaction @'Shelley w
+        signedTx <- getFromResponse (#transaction . #getApiT . #serialisedTx) <$>
+            request @ApiSignedTransaction ctx signEndpoint Default toSign
+
+        -- Submit tx
+        let submitEndpoint = Link.postExternalTransaction
+        let headers = Headers
+                [ ("Content-Type", "application/octet-stream")
+                , ("Accept", "application/json")
+                ]
+        r <- request @ApiTxId ctx submitEndpoint headers (NonJson $ BL.fromStrict signedTx)
+        verify r
+            [ expectResponseCode HTTP.status202
+            ]
+
+    it "TRANS_NEW_SIGN_02 - Rejects unsigned transaction" $ \ctx -> runResourceT $ do
+        w <- fixtureWallet ctx
+        let amt = minUTxOValue (_mainEra ctx)
+
+        -- Construct tx
+        payload <- mkTxPayload ctx w amt
+        let constructEndpoint = Link.createUnsignedTransaction @'Shelley w
+        sealedTx <- getFromResponse (#transaction . #getApiT . #serialisedTx) <$>
+            request @(ApiConstructTransaction n) ctx constructEndpoint Default payload
+
+        -- Submit tx
+        let submitEndpoint = Link.postExternalTransaction
+        let headers = Headers
+                [ ("Content-Type", "application/octet-stream")
+                , ("Accept", "application/json")
+                ]
+        r <- request @ApiTxId ctx submitEndpoint headers (NonJson $ BL.fromStrict sealedTx)
+        verify r
+            [ expectResponseCode HTTP.status500
```

Comment:
Perhaps it would be nice to "catch" `500` and have some `40x` here?

Reply:
We thought the same, though not really relevant to this PR but I think the submitEndpoint should not return 500 errors for invalid transactions, since this is precisely a client error.

Reply:
I've issued bug ticket for this ADP-1172.